scripts/release_dev.js

```diff
@@ -0,0 +1,111 @@
+const path = require('path');
+
+const semver = require('semver');
+
+const {
+  askForConfirmation,
+  askForText,
+  build,
+  checkGitStatus,
+  checkNpmAuth,
+  getDateYYYYMMDDHHMMSS,
+  revertCommit,
+  runExec,
+} = require('./release.helpers');
+
+const packageJsonPath = path.resolve(__dirname, '../package.json');
+const packageJson = require(packageJsonPath);
+
+async function checkGitBranchAndStatus() {
+  console.log('Checking your PR branch...');
+  const resultBranch = await runExec('git branch --show-current', {
+    silent: true,
+  });
+  const branchName = resultBranch.stdout.toString().trim();
+  if (branchName === 'main') {
+    console.error(`🟠 You are at "main". Are you sure you wanna release a dev version here?`);
+    process.exit(1);
+  }
+
+  await checkGitStatus();
+}
+
+async function getNewVersion() {
+  const hour = getDateYYYYMMDDHHMMSS(); // Use hour to be an incremental number
+  const currentVersion = packageJson.version;
+
+  if (currentVersion.includes('-dev.')) {
+    console.log('Bumping exisiting dev...');
+    return currentVersion.replace(/-dev\.(.*)$/, `-dev.${hour}`);
+  }
+
+  console.log('Creating a new dev...');
+  const versionType = process.argv.slice(2)[0]; // major | minor | patch
+  if (!versionType) {
+    console.log('🟠 version type is missing. Make sure to run the script from package.json');
+    process.exit(1);
+  }
+  const versionBase = semver.coerce(currentVersion); // 1.0.0-xxxx -> 1.0.0
+  return semver.inc(versionBase, versionType) + `-dev.${hour}`;
+}
```

Comment:
In case you were wondering why the commit SHA in the dev release version wasn't matching the current commit at the time the script is executed, this is why: https://www.loom.com/share/d5a02b42b04b407195e2b30311ba32c1

Reply:
Nice TIL but you missed the point I want the SHA of the next commit (not existing yet). For example:

```bash
- 19ea0c1 "Release v0.1.0-dev.19ea0c1"
- 575f1a0 "feat: add potatoes"
- 9e21c3 "feat: add potatoes"
````

I need to know "19ea0c1" before I commit it so it matches in the commit msg.

But either way, even  if we find the answer, i'd rather stick with the date hash (yyyy..ss) because it's chronological and it was already useful yesterday to verify some versions.

Reply:
Ah got it! Yeah I like the date-based version, that's a good idea 💡